common/web/input-processor/tests/cases/predictionContext.js

```diff
@@ -71,22 +71,20 @@ describe("PredictionContext", () => {
     await langProcessor.loadModel(appleDummyModel);  // await:  must fully 'configure', load script into worker.
 
     const kbdProcessor = new KeyboardProcessor(deviceSpec);
+    const predictiveContext = new PredictionContext(langProcessor, kbdProcessor);
+
+    let updateFake = sinon.fake();
+    predictiveContext.on('update', updateFake);
 
     let mock = new Mock("appl", 4); // "appl|", with '|' as the caret position.
     const initialMock = Mock.from(mock);
-
-    let context = new PredictionContext(langProcessor, kbdProcessor, mock);
-
-    let updateFake = sinon.fake();
-    context.on('update', updateFake);
-    context.sendUpdateEvent(); // Allows external code to request a re-retrieval of current suggestion state.
+    const promise = predictiveContext.setCurrentTarget(mock);
```

Comment:
A bit of happy fallout - I think this feels cleaner and simpler when it comes to unit testing than the prior pattern.